lib/Sema/CSDiagnostics.cpp

```diff
@@ -6760,3 +6760,19 @@ void TrailingClosureRequiresExplicitLabel::fixIt(
   diagnostic.fixItInsert(newRParenLoc,
                          isExpr<SubscriptExpr>(anchor) ? "]" : ")");
 }
+
+bool InvalidEmptyKeyPathFailure::diagnoseAsError() {
+  auto *KPE = getAsExpr<KeyPathExpr>(getAnchor());
+  assert(KPE && KPE->hasSingleInvalidComponent() &&
+         "Expected a malformed key path expression");
+
+  // String interpolation represented as key path expressions e.g. \(x),
+  // \(x, a: 1) and this is already diagnosed. So let's skip it because
+  // that is not the case for a empty key path diagnostic.
```

Comment:
How about “If we have a string interpolation that is represented as a key path, such as \(x) or \(x, a: 1), then bail out because it would’ve already been diagnosed.”

Reply:
Fixed! Thanks :)